### DISCUSSION TITLE
Add polite CTA to articles

### DISCUSSION THREAD

**Discussion by laurelsilva** [eggheadio/egghead-next#824] [Pitches]
**Articles have ZERO CTAs.** Adding CTA's to articles is essential to customer acquisition and retention. A good first step is adding the "Polite" multi-stage CTA component to the articles. It is polite and inviting when reading an article. It's multi-stage and gives the user the opportunity to "eject" at any time.

[Live Example](https://escuelafrontend.com/articulos/la-guia-definitiva-de-console-api)

https://user-images.githubusercontent.com/57044804/129416921-57573289-c543-4495-a3e2-bf761a794e24.mp4

## Implementation Details

The component already exists in the [Skill Recordings Starter](https://github.com/skillrecordings/next-skill-product-starter/blob/main/src/components/forms/convertkit/polite/index.tsx) and it has been implemented in several [partner products](https://github.com/skillrecordings/escuelafrontend.com/blob/main/src/components/forms/convertkit/polite/index.tsx). 

The current implementation of the component uses Convertkit for the subscribe form. We will be using [Customer.io](http://customer.io) instead and summiting the form to the ["New Learner"](https://fly.customer.io/env/97673/campaigns/18/overview?channels=email_twilio_webhook_slack) campaign.

We can create a new subscriber form using the [`EmailEntryForm`](https://github.com/eggheadio/egghead-next/blob/2cb7ee541397c2af6adb83207b694da859352cf6/src/components/customer-io/email-entry-form.tsx) component.

[Own your Online Presence](https://egghead.io/own-your-online-presence) is an example that uses the customer.io `EmailEntryFrom` component. 

Are there any additonal considerations missing here?

## Copy

Pending TODO on the actual copy of the CTA

**Comment by joelhooks** [comment]
I think this switch statement is relevant here as a sort of "proto-plinko"

https://github.com/eggheadio/egghead-next/blob/main/src/components/app/header/index.tsx#L48-L73

![](https://c.tenor.com/w3bN4E1l_fQAAAAd/plinko-win.gif)

So we can take a look at converkit, customer-io, and egghead data and determine what is the appropriate CTA to present to the current learner. This can start very simply!

**Comment by joelhooks** [comment]
related #826 discussion about the "global CTA plinko"

**Comment by MaggieAppleton** [comment]
In terms of the UX design of the component have two small notes:

1. If I type in an incorrect email address the error message shows up in english instead of spanish:
![image](https://user-images.githubusercontent.com/5599295/129537064-53f9cc41-064a-45ac-b870-78affb2ef615.png)
We should double check all error messages are in Spanish so it's consistent across the site

2. I expected that clicking "no" would make the banner go away. Instead it did nothing. In general if we put a button on a screen it needs to be clickable and trigger an action.
![image](https://user-images.githubusercontent.com/5599295/129537046-22931547-5b30-439b-b529-9deff631f0e8.png)
I'd recommend making "no" trigger the same action as clicking the small x in the corner – dismisses the popup

**Reply by laurelsilva** [reply]
@MaggieAppleton good catch on both points!

**Comment by MaggieAppleton** [comment]
Re: plinko component. If we're going to build this as a reusable component and already getting the user to answer questions on an in-app pop-up, we may as well create the in the functionality to ask them multiple questions before delivering a CTA.

Eg. Do you want to improve you frontend skills? Yes / no --> What level are you at in your career? Just starting out / Junior / Mid-level / Senior --> Want our {introductory/advanced} series on {React/JavaScript}?

No sure how much that suggestion complicates matters.
